src/citrine/informatics/data_sources.py

```diff
@@ -92,13 +136,37 @@ class GemTableDataSource(Serializable['GemTableDataSource'], DataSource):
     table_id = properties.UUID("table_id")
     table_version = properties.Integer("table_version")
 
+    _data_source_type = "gemd"
+
     def __init__(self,
                  *,
                  table_id: UUID,
                  table_version: Union[int, str]):
         self.table_id: UUID = table_id
         self.table_version: Union[int, str] = table_version
 
+    @classmethod
+    def _data_source_id_builder(cls, *args) -> DataSource:
+        return GemTableDataSource(table_id=UUID(args[0]), table_version=args[1])
+
+    def to_data_source_id(self) -> str:
+        """Generate the data_source_id for this DataSource."""
+        return "::".join(
+            str(x) for x in [self._data_source_type, self.table_id, self.table_version]
+        )
```

Comment:
Nitpick: what's the benefit of this over an f-string?

f`{self._data_source_type}::{self.table_id}::{self.table_version}`

Reply:
My thought when writing was that we have a series of identifiers joined by `::` in all cases, but yes, it ended up not great.  I thought of creating a `get_identifiers` method that got joined at the parent level, but then that seemed a bit absurd.  I'll swap to f-string.